source/extensions/transport_sockets/tls/cert_validator/cert_validator.h

```diff
@@ -43,7 +43,9 @@ struct ValidationResults {
 class CertValidator {
 public:
   // Wraps cert validation parameters added from time to time.
-  struct ExtraValidationContext {};
+  struct ExtraValidationContext {
+    absl::string_view host_name;
```

Comment:
> Do we ever validate a certificate without a hostname being present? 

We have been doing that for ssl transport socket. But if we use auto_sni, there is a filter state keyed with `Network::UpstreamSubjectAltNames` to specify sni list (from `Host` header) to override the ones in the config. And the overriding values get verified against the leaf cert in DefaultCertValidator. 

> I wonder if it would make sense to pass this in to doVerifyChain directly as an explicit argument?

done